src/gpu/jit/ngen/ngen_utils.hpp

```diff
@@ -19,7 +19,9 @@
 
 #include <cstdint>
 
+#ifdef __INTEL_COMPILER
```

Comment:
Since `immintrin.h` is not Intel Compiler specific but rather X86-64 specific the condition seems to be incorrect. We do use the condition in other places but it's probably because we want to use intrinsics only with Intel Compiler.
The right condition would be `#if defined(__x86_64__) || defined(_M_X64)`.

Also, some functionality from the header must be used somewhere so excluding the header without guarding those places should cause issues at build time. Does the code compile when the header is excluded?

Reply:
Yes, the code compile with the code been excluded that way.
So indeed, It might be possible to remove the include... (tested with gcc on fedora)

Reply:
I could not find any intrinsics from that header used in the ngen code. Likely the intrinsics were later replaced with the `__builtin_*`  functions used in `bsf`, `bsr`, and `popcnt` functions so it is now unneeded.

Reply:
Alright, let's remove it then.

Reply:
Forced pushed a new commit to drop the immintrin.h header instead.

Reply:
@kwizart I'll pull this PR in by end of this week.